assets/wire/reducers.js

```diff
@@ -210,19 +210,52 @@ export default function wireReducer(state = initialState, action) {
             bookmarkedItems: state.bookmarkedItems.filter((val) => val !== action.item),
         };
 
-    case SET_NEW_ITEMS:
+    case SET_NEW_ITEMS: {
+        const newItemsByTopic = state.newItemsByTopic || {};
+        action.data.topics.map((topic) => {
+            newItemsByTopic[topic] = newItemsByTopic[topic] || [];
+            newItemsByTopic[topic].push(action.data.item);
+        });
+
+        const newItems = state.newItems || [];
+        newItems.push(action.data.item);
+
+        const headerNotification = document.getElementById('header-notification');
+        if (headerNotification){
+            headerNotification.innerHTML = newItems.length > 0 ? newItems.length : '';
+        }
+
         return {
             ...state,
-            newItemsCount: action.newItems.length,
-            newItemsData: action.newItems.length ? action.data : null,
+            newItems,
+            newItemsByTopic
         };
+    }
 
-    case REFRESH_ITEMS: {
-        const nextState = recieveItems(state, state.newItemsData);
-        return Object.assign(nextState, {
-            newItemsCount: 0,
-            newItemsData: null,
-        });
+
+    case REMOVE_NEW_ITEMS: {
+        const newItemsByTopic = state.newItemsByTopic || {};
+        let newItems = state.newItems || [];
+
+        if (state.newItemsByTopic[action.data]) {
+
+            newItemsByTopic[action.data].map((item) => {
+                newItems = newItems.filter((newItem) => newItem._id !== item._id);
+            });
+        }
+
+        newItemsByTopic[action.data] = null;
+
+        const headerNotification = document.getElementById('header-notification');
```

Comment:
this part shouldn't be here, but you can add it to wire app and use [portal](https://reactjs.org/docs/portals.html)